addons/website/static/src/builder/plugins/options/avatars_option_plugin.js

```diff
@@ -0,0 +1,97 @@
+import { Plugin } from "@html_editor/plugin";
+import { registry } from "@web/core/registry";
+import { BuilderAction } from "@html_builder/core/builder_action";
+import { renderToElement } from "@web/core/utils/render";
+import { AvatarsHeaderMiddleButtons } from "./avatars_header_buttons";
+
+function _updateZIndex(el) {
+    const avatarEls = el.querySelectorAll(".o_avatar");
+    if (!el.classList.contains("o_first_on_top")) {
+        avatarEls.forEach(avatarEl => {
+            avatarEl.style.setProperty("--avatar-z-index", "");
+        });
+        return;
+    }
+    const totalAvatars = avatarEls.length;
+    avatarEls.forEach((avatarEl, index) => {
+        avatarEl.style.setProperty("--avatar-z-index", totalAvatars - index);
+    });
+}
+
+class AvatarsOptionPlugin extends Plugin {
+    static id = "avatarsOption";
+    static dependencies = ["builderOptions"];
+
+    resources = {
+        builder_options: [
+            {
+                template: "website.AvatarsOption",
+                selector: ".s_avatars",
+                cleanForSave: _updateZIndex,
+            },
+        ],
+        builder_header_middle_buttons: {
+            Component: AvatarsHeaderMiddleButtons,
+            selector: ".s_avatars",
+            props: {
+                addAvatar: this.addAvatar.bind(this),
+                removeAvatar: this.removeAvatar.bind(this),
+            },
+        },
+        builder_actions: {
+            AvatarsAddContentAction,
+        },
+        so_content_addition_selector: [".s_avatars"],
+        is_unremovable_selector:
+            ".s_avatars_rules_collector .o_avatar:first-of-type:last-of-type",
+    };
+
+    addAvatar(editingElement) {
+        const newAvatarIndex = editingElement.querySelectorAll('.o_avatar').length + 1;
+        const picIndex = Math.min(newAvatarIndex, 6);
+        const newAvatarEl = renderToElement("website.s_avatars.avatar", { picIndex });
```

Comment:
You can therefore remove the "index picking" code (see comment on the `s_avatars.avatar` template) :slightly_smiling_face: